### PR TITLE
Make sure estimateClaim is accurate

### PR DESCRIPTION
**Pull request by alexcos20**
Iterate over 52 weeks, for multiple actors and cases, to make sure that estimateClaim is accurate

### CODE REVIEW

File: contracts/ve/FeeEstimate.vy
Comment (trentmc) [inline contracts/ve/FeeEstimate.vy:98]: What's going on here? The expression "x / K * K" simplifies to "x"

**Review by trentmc — APPROVED** [review]
In general looks good. 

But, please fix (or document) the weird multiply - times expression before merging.

File: util/test/veOcean/test_rewards.py
Comment (trentmc) [inline util/test/veOcean/test_rewards.py:70]: If all this code is commented out, why have it at all?

**Comment by alexcos20** [comment]
for now, this is just a test branch, where I can play and share code.  Pls do not merge

**Comment by trentmc** [comment]
> for now, this is just a test branch, where I can play and share code. Pls do not merge

Thanks. OK. I'd got a request to review it, hence why I was giving review comments.

I've now marked it as draft, and at the top of the PR, I've marked it as a play / test branch. All clear! Pura vida :)

File: contracts/ve/FeeEstimate.vy
Comment (alexcos20) [reply contracts/ve/FeeEstimate.vy:98]: That was my original thought as well, but actually it's round down :)  This is EVM, so everything is integer.

File: util/test/veOcean/test_rewards.py
Comment (alexcos20) [reply util/test/veOcean/test_rewards.py]: fixed

**Comment by alexcos20** [comment]
Cleaned up the code, tests looks good now.  Ready for review

**Comment by alexcos20** [comment]
Again, purpose of this test is to make sure that veFeeEstimate.estimateClaim()  equals with veFeeDistributor.claim()

It does not check it veFeeDistributor.claim() is accurate (user is receiving what he should, according to rewards, vote_power, etc..)

File: contracts/ve/FeeEstimate.vy
Comment (trentmc) [reply contracts/ve/FeeEstimate.vy:98]: Thanks. I was guessing it was something like that. And I believe it's round down, at the resolution of one week.

We should put a comment above, to that effect.

**Review by trentmc — CHANGES_REQUESTED** [review]
Just one small tweak requested. Then I'm happy to approve

**Comment by alexcos20** [comment]
@trentmc - see above commit

**Comment by alexcos20** [comment]
In the above commit, I have added extra checkpoint checks in veFeeEstimate..

In https://github.com/oceanprotocol/df-py/blob/bug/fix_veFeeEstimate/contracts/ve/FeeDistributor.vy#L311-L318 , there are some checkpoints that are being executed if needed.  Since veFeeEstimate is not able to run those checkpoints, we should return 0, because most likely, the estimate will not be accurate  (since estimate will run on old checkpoints and veFeeDistributor.claim will update the checkpoint)

@trizin @trentmc @idiom-bytes, thoughts ?

File: util/test/veOcean/test_rewards.py
Comment (alexcos20) [reply util/test/veOcean/test_rewards.py]: done

**Comment by alexcos20** [comment]
so, @trizin , all good? can we merge it ?

**Pull request merged**
